### PR TITLE
Induction principle for smash product

### PR DESCRIPTION
**Pull request by aljungstrom**
Contains:

- New induction principle for smash products: `⋀̃→⋀-ind` in `Cubical.HITs.SmashProduct.Induction`
- Changed of a typo in `Cubical.HITs.SmashProduct.Base` ('3' to '2')
- Added finite families of types (needed for first point)
- Updated related paper file.

### CODE REVIEW

File: Cubical/Data/Fin/Base.agda [hunk 1]
```diff
@@ -109,3 +113,29 @@ FinPathℕ : {n : ℕ} (x : Fin n) (y : ℕ) → fst x ≡ y → Σ[ p ∈ _ ] (
 FinPathℕ {n = n} x y p =
     ((fst (snd x)) , (cong (λ y → fst (snd x) + y) (cong suc (sym p)) ∙ snd (snd x)))
   , (Σ≡Prop (λ _ → isProp≤) p)
+
+FinVec : (A : Type ℓ) (n : ℕ) → Type ℓ
+FinVec A n = Fin n → A
```
Comment (mortberg) [inline Cubical/Data/Fin/Base.agda:118]: This is not a vector?

File: Cubical/HITs/SmashProduct/Induction.agda [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+⋀̃→⋀ : ⋀̃ Aᵢ → ⋀ᵢ Aᵢ
+
+where ⋀̃ is defined as the pushout
+
+Σi≤n (A₀ × ... × Aᵢ₋₁ × Aᵢ₊₁ × ... × Aₙ) ---→ (A₀ × ... × Aₙ)
+               ↓                                ⌟  ↓
+               1 ------------------------------→ ⋀̃ Aᵢ
+
+Statment: Let f,g : ⋀ᵢ Aᵢ → B. If f ∘ ⋀̃→⋀ = g ∘ ⋀̃→⋀, then f = g.
```
Comment (mortberg) [inline Cubical/HITs/SmashProduct/Induction.agda:14]: Typo (statment)

**Review by mortberg — CHANGES_REQUESTED** [review]
Two small comments then we can merge

File: Cubical/Data/Fin/Base.agda (hunk 1 above)
Comment (aljungstrom) [reply Cubical/Data/Fin/Base.agda:118]: I just copied the naming in the other definition of Fin (FinData):

https://github.com/agda/cubical/blob/c6a96eeefc6f5f1ce346512d3130de45e4b9a065/Cubical/Data/FinData/Base.agda#L76

Isn't it reasonable-ish?

Comment (mortberg) [reply Cubical/Data/Fin/Base.agda:118]: Yes

**Review by mortberg — APPROVED** [review]

**Pull request merged**